taxonomy/models.py

```diff
@@ -102,6 +102,119 @@ class Meta:
         app_label = 'taxonomy'
 
 
+class XblockSkills(TimeStampedModel):
+    """
+    Skills that will be learnt by completing xblock.
+
+    .. no_pii:
+    """
+
+    usage_key = models.CharField(
+        unique=True,
+        max_length=255,
+        help_text=_('The key of the xblock whose text was used for skills extraction.')
+    )
+    skills = models.ManyToManyField(
+        Skill,
+        through="XblockSkillThrough",
+        help_text=_(
+            'The ID of the skill extracted for the xblock.'
+        )
+    )
+    requires_verification = models.BooleanField(
+        default=True,
+        help_text=_('Indicates whether skills applied to this block requires verification from users'),
+    )
+    auto_processed = models.BooleanField(
+        default=False,
+        help_text=_('Indicates whether the text from this block was already processed'),
+    )
+
+    class Meta:
+        """
+        Meta configuration for CourseSkills model.
+        """
+
+        verbose_name = 'Xblock Skills'
+        verbose_name_plural = 'Xblock Skills'
+        ordering = ('created', )
+        app_label = 'taxonomy'
+
+    def __str__(self):
+        """
+        Create a human-readable string representation of the object.
+        """
+        return '<XblockSkills usage_key="{}">'.format(self.usage_key)
+
+    def __repr__(self):
+        """
+        Create a unique string representation of the object.
+        """
+        return '<XblockSkills id="{0}">'.format(self.id)
+
+
+class XblockSkillThrough(TimeStampedModel):
+    """
+    Skills that will be learnt by completing xblock.
+
+    .. no_pii:
+    """
+
+    skill = models.ForeignKey(Skill, on_delete=models.CASCADE)
+    xblock = models.ForeignKey(XblockSkills, on_delete=models.CASCADE)
+    relevant_count = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text=_('Number of times learners verified this skill')
```

Comment:
How do we stop learners from voting twice? Is there also a table of learners that have voted for each XBlock?

Reply:
We will have a flag in xblock like so.

![image](https://user-images.githubusercontent.com/10894099/199909727-f62c1f44-c90c-4b1d-92e8-3dd4b9f4e8fb.png)